canvas_modules/common-canvas/src/toolbar/toolbar.scss

```diff
@@ -18,131 +18,194 @@
-	position: fixed;
-	z-index: 2;
+	display: flex;
+	justify-content: space-between;
+
+	/* Prevent elements from being dragged onto canvas */
+	-webkit-user-drag: none;
 }
 
-#toolbar-items {
-	list-style-type: none;
-	margin: 0;
-	padding: 0;
+.toolbar-left-bar {
+	flex-shrink: 5000000; // Make sure left bar always shrinks first
+	display: block;
+	padding-right: 42px;
+	overflow: hidden;
+	align-items: center;
 
-	display: flex;
 }
 
-.toolbar-items-container {
+.toolbar-right-bar {
 	display: flex;
+	flex-direction: row-reverse;
+	flex-wrap: wrap;
+	justify-content: flex-start;
+	flex-shrink: 1;
+	overflow: hidden;
 }
 
-#zoom-actions-container {
-	position: absolute;
-	right: 2px; // accommodate notification onClick border
-}
+.toolbar-item {
+	background-color: $ui-01;
+	display: inline-block;
+
+	& button {
+		padding: 0;
+		border-top: 0;
+		border-bottom: 0;
+		min-height: 30px;
+	}
 
-.list-item-containers {
-	&.notificationCounterIcon {
-		.toolbar-item {
-			position: relative;
+	&.default,
+	&.ghost {
+		& button {
+			background-color: $ui-01;
 		}
-		.text-content {
-			position: absolute;
-			z-index: 2;
-			top: 13px;
-			font-size: 9px;
-			line-height: 9px;
-			color: $icon-01;
+
+		& button:hover {
+			background-color: $hover-field;
 		}
-	}
 
-	.list-item {
-		padding: 0;
-		min-height: $toolbar-button-width;
-		width: $toolbar-button-width;
-		display: flex;
-		justify-content: center;
-		user-select: none;
-
-		&.overflow {
-			height: auto;
-			width: 200px;
-			display: block;
+		& button:disabled:hover {
+			background-color: $ui-01;
 		}
 	}
 
-	.list-item-disabled svg {
-		fill: $disabled-02;
+	&.tertiary {
+		& button {
+			color: $brand-03;
+			background-color: $ui-01;
+		}
+
+		& button:hover {
+			background-color: $hover-field;
+		}
+
+		& button:disabled,
+		& button:disabled:hover {
+			color: $disabled-02;
+			background-color: $ui-01;
+		}
 	}
 }
 
-.toolbar-divider {
-	border-right: $toolbar-divider-width solid $ui-03;
+.toolbar-spacer {
+	width: 0;
+	display: inline-block;
 }
 
-.overflow-toolbar-divider {
-	border-bottom: $toolbar-divider-width solid $ui-03;
+.toolbar-overflow-item {
+	background-color: $ui-01;
+	display: inline-block;
+
+	& button {
+		padding: 0;
+		border-top: 0;
+		border-bottom: 0;
+		min-height: 30px;
+		background-color: $ui-01;
+		border-right: $toolbar-divider-width solid $ui-03;
+		// Reduce width by 1 to hide the right border for the rightmost overflow icon in the leftbar.
+		width: calc(#{$toolbar-button-width} - 1px);
+	}
+
+	& button:hover {
+		background-color: $hover-field;
+	}
+
+	& button:disabled:hover {
+		background-color: $ui-01;
+	}
 }
 
-.toolbar-item {
-	// set height/width for tooltip
-	height: 34px;
-	width: 34px;
+.toolbar-overflow-menu-item button {
+	width: 100%;
+	padding-top: 2px;
+	padding-bottom: 2px;
+	border: 0;
+	min-height: 20px;
+}
 
+.toolbar-item-content {
+	width: inherit;
 	display: flex;
 	justify-content: center;
+	padding: 12px 12px 8px;
```

Comment:
Can you change this to `padding: 11px`? This will help center all the icons. More visibly with the counter in notification icon. With this, we should also set the `.toolbar-icon` to `height: 16px;`

Reply:
Done